library/pk.c

```diff
@@ -853,6 +854,88 @@ mbedtls_pk_type_t mbedtls_pk_get_type(const mbedtls_pk_context *ctx)
 }
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
+int mbedtls_pk_get_ec_public_key_props(mbedtls_pk_context *pk,
+                                       psa_ecc_family_t *ec_curve, size_t *bits)
+{
+    if ((pk == NULL) || (ec_curve == NULL) || (bits == NULL)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+    if ((pk->pk_ec_family == 0) ||
+        (pk->pk_bits == 0)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+
+    *ec_curve = pk->pk_ec_family;
+    *bits = pk->pk_bits;
+
+    return 0;
+}
+
+#if defined(MBEDTLS_ECP_LIGHT)
+int mbedtls_pk_update_public_key_from_keypair(mbedtls_pk_context *pk)
+{
+    int ret = MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE;
+    mbedtls_ecp_keypair *ecp_keypair;
+
+    if (pk == NULL) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+    /* RSA does not support raw public keys inside the pk_context structure,
+     * so we quit silently in this case */
+    if ((pk->pk_info->type != MBEDTLS_PK_ECKEY) &&
+        (pk->pk_info->type != MBEDTLS_PK_ECKEY_DH) &&
+        (pk->pk_info->type != MBEDTLS_PK_ECDSA)) {
+        return 0;
+    }
+
+    ecp_keypair = mbedtls_pk_ec(*pk);
+
+    ret = mbedtls_ecp_point_write_binary(&ecp_keypair->grp, &ecp_keypair->Q,
+                                         MBEDTLS_ECP_PF_UNCOMPRESSED,
+                                         &pk->pk_raw_len,
+                                         pk->pk_raw,
+                                         MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN);
+    if (ret != 0) {
+        return ret;
+    }
+
+    pk->pk_ec_family = mbedtls_ecc_group_to_psa(ecp_keypair->grp.id,
+                                                &pk->pk_bits);
+
+    return 0;
+}
+
+int mbedtls_pk_update_keypair_from_public_key(mbedtls_pk_context *pk)
```

Comment:
Note: it doesn't look like this function is currently used anywhere (from grepping the source). On one hand, perhaps it will be useful later, but OTOH, not used means untested... Wdyt?